### ISSUE TITLE
⌨ Apply to this recruitment button does not have assigned item numbers

### ISSUE THREAD

**Issue by RM-1978** [GCTC-NTGC/gc-digital-talent#5358]
## ❌ Current behaviour (issue)
**Apply to this recruitment** button does not have assigned item numbers for alternative navigation. 
### Location: 
Pool Advertisement (Job Poster)
### Category: 
Critical Accessiblity Bug
### AT configuration(s):  
Voice Control, Mobile, IOS, Chrome
Reproduced with TBS iPhone and Voice Control by browsing to server of dev workstation
### How to reproduce
**With Voice Control** 
1.	Navigate to the browse opportunities page
2.	Choose numbered grid overlay. 
3.	Observe there is no number available to tap the "Apply to this recruitment" button
### Fable Video Clip:
[Signature interaction Issues](https://app.makeitfable.com/clip/63bcd337604482001abf7428) 

## :camera: Screenshot
![20230126_085649](https://user-images.githubusercontent.com/8978655/214858878-12d3d24d-3c89-45a2-a0cc-7028568e2700.jpg)

## ✔ Expected behaviour (fix)
The user can activate the Apply to this recruitment button and the Job Poster details page loads. 
**Potential WCAG Considerations**  [ 2.1.1 Keyboard Navigation](https://www.w3.org/WAI/WCAG22/Understanding/keyboard).

### Acceptance Criteria

- [ ] **Apply to this recruitment** buttons have assigned item numbers 
- [ ] Button is accessible for alternative navigation

### :exclamation: Bonus!
If you think this solution can fix  #5357 as well, pull it in as an interruption.

**Comment by esizer** [comment]
Every time I try to take a screen shot the numbers go away but I have tested in both macOS and windows and they appear for me 😕

**Comment by petertgiles** [comment]
@RM-1978 Could you please try to find the exact versions of the tools used in this test to help us reproduce?

**Comment by petertgiles** [comment]
Too bad.  :disappointed:   Is there any possibility that you could try to reproduce it sometime?

**Comment by RM-1978** [comment]
Hi peer, Don<t know if seeing the Fable user produce the issue might help 
[Signature interaction Issues](https://app.makeitfable.com/clip/63bcd337604482001abf7428)

**Comment by petertgiles** [comment]
I managed to replicate the "apply" button issue on my TBS iPhone!  I can't easily sign-in so I'll split off the signature issue and ask Gray when we can open UAT to testing again.

**Comment by patcon** [comment]
@gobyrne noted that this might have something to do with button being on page twice (same role and same text and perhaps not `id` attribute). If so, added something distinguishing might help the screenreader resolve the differences between the buttons

**Comment by petertgiles** [comment]
I think I've found the problem but I'm not sure how to fix it. 

We use CSS to hide extra context for AT users on these links `data-h2-visually-hidden="base(invisible)"`.
![image](https://user-images.githubusercontent.com/8978655/215105790-5a31d749-8411-4daa-bedc-b40b8453e733.png)


It seems like this hidden text is what breaks the Voice Control numbering.  I tested some configurations of this button with screenshots below.

With hidden context text (currently in production):
![20230127_090005](https://user-images.githubusercontent.com/8978655/215105953-53fd854f-030f-424c-9978-2135c9bb8f69.jpg)

With context not-hidden:
![20230127_085914](https://user-images.githubusercontent.com/8978655/215105949-0dafa905-fdf3-4c72-9e11-5b9c9f15c3d7.jpg)

With context removed:
![20230127_090103](https://user-images.githubusercontent.com/8978655/215105955-229956f5-a461-4a9b-a616-5341efe8ab14.jpg)

**Comment by esizer** [comment]
> I think I've found the problem but I'm not sure how to fix it.
> 
> We use CSS to hide extra context for AT users on these links `data-h2-visually-hidden="base(invisible)"`.

Now that you have done the hard work to figure this out, I am not surprised this is the issue. We have other known issues with VoiceOver and our hidden text 😢 

I'm not sure the best way to handle this other than showing the text 🤷‍♀️ We need it to follow the [links with identical accessible name has same purpose rule](https://dequeuniversity.com/rules/axe/4.1/identical-links-same-purpose). 

We could use `aria-label` but that would require two strings for translation. The other solution would be requesting link text be crafted in a way that we do not need to rely on hidden text or `aria-label`.

My preference would be to use custom link text. I'm not a **_huge_** fan of relying on hidden text in any scenario myself.

**Comment by petertgiles** [comment]
Yeah, I'll bring it up in scrum today.   I was doing some reading today and one site recommended against hidden text for the reason that you can't say "Tap Apply to this recruitment" because it's expecting you to say the full accessible text with the hidden part.

**Comment by petertgiles** [comment]
short term solution agreement: add classification in parenthesis without hidden text
